pkg/services/pluginsettings/service.go

```diff
@@ -12,16 +12,35 @@ import (
 	"github.com/grafana/grafana/pkg/services/sqlstore"
 )
 
-type ServiceImpl struct {
-	Bus            bus.Bus
-	SQLStore       *sqlstore.SQLStore
-	SecretsService secrets.Service
+func ProvideService(bus bus.Bus, store *sqlstore.SQLStore, secretsService secrets.Service) *Service {
+	s := &Service{
+		bus:            bus,
+		sqlStore:       store,
+		secretsService: secretsService,
+		logger:         log.New("pluginsettings"),
+		pluginSettingDecryptionCache: secureJSONDecryptionCache{
+			cache: make(map[int64]cachedDecryptedJSON),
+		},
+	}
+
+	s.bus.AddHandler(s.GetPluginSettingById)
+	s.bus.AddHandler(s.UpdatePluginSetting)
+	s.bus.AddHandler(s.UpdatePluginSettingVersion)
+
+	return s
+}
+
+type Service struct {
+	bus            bus.Bus
+	sqlStore       *sqlstore.SQLStore
+	secretsService secrets.Service
 
 	logger                       log.Logger
 	pluginSettingDecryptionCache secureJSONDecryptionCache
 }
 
-type Service interface {
+type Store interface {
```

Comment:
I think in enterprise and the way we will be splitting sqlstore, this layer will be called Manager - instead of Store or Service, because we will split the sqlstore and create a plugin settings Store